examples/events_custom_state_event/event.rs

```diff
@@ -0,0 +1,33 @@
+use amethyst::{
+    core::{
+        shrev::{EventChannel, ReaderId},
+        EventReader,
+    },
+    derive::EventReader,
+    ecs::{Read, SystemData, World},
+};
+use amethyst_input::{BindingTypes, InputEvent, StringBindings};
+use amethyst_ui::UiEvent;
+use winit::Event;
+
+/// Here's a copy of the original StateEvent with our own type added
+#[derive(Clone, Debug, EventReader)]
+#[reader(MyExtendedStateEventReader)]
+pub enum MyExtendedStateEvent<T = StringBindings>
+where
+    T: BindingTypes + Clone,
+{
+    /// Events sent by the winit window.
+    Window(Event),
+    /// Events sent by the ui system.
+    Ui(UiEvent),
+    /// Events sent by the input system.
+    Input(InputEvent<T>),
+    /// Our own events for our own game logic
+    MyCustomGameEvent(MyCustomGameEventType),
```

Comment:
nit: Can we use realistic variant names? I find placeholder names can make things harder to grasp